runtime/compiler/optimizer/EscapeAnalysis.hpp

```diff
@@ -645,6 +650,15 @@ class TR_EscapeAnalysis : public TR::Optimization
    TR_ScratchList<TR_DependentAllocations> _dependentAllocations;
    TR_BitVector *             _vnTemp;
    TR_BitVector *             _vnTemp2;
+
+   typedef TR::typed_allocator<std::pair<TR::Node* const, TR_Array<TR::Node*>*>, TR::Region&> NodeToNodeArrayMapAllocator;
+   typedef std::less<TR::Node*> NodeComparator;
+   typedef std::map<TR::Node*, TR_Array<TR::Node*>*, NodeComparator, NodeToNodeArrayMapAllocator> NodeToNodeArrayMap;
+
+   NodeToNodeArrayMap *       _nodeUsesThroughAternary;
+
+   friend class TR_FlowSensitiveEscapeAnalysis;
```

Comment:
are the declarations of these brought in from the header includes? if not a forward declaration in this header is a must to avoid undefined behaviour on whether a friend class declaration is a forward declaration as well or not.

Reply:
Fixed in commit https://github.com/eclipse/openj9/pull/5699/commits/f822f8ae1f3604b2a5c3a855d3476f2390a9c4e0